### DISCUSSION TITLE
Issues running local build of React-Aria

### DISCUSSION THREAD

**Discussion by bu3alwa** [adobe/react-spectrum#6307] [Q&A]
There might be something I am missing but here is the steps I have done to get started.

```
# on master branch
nvm use
yarn install
yarn run build

# on nextjs app
npm install ../../../react-spectrum/packages/react-aria
npm install ../../../react-spectrum/packages/react-aria-components
npm install ../../../react-spectrum/packages/@react-aria/utils
```

![image](https://github.com/adobe/react-spectrum/assets/3469062/96a8c6d8-f3d4-43e9-a4c4-f246fb390a89)

I seem to be running into runtime errors like this one or one where it was running into issues subscribing on a `useSyncExternalStore`.

Any ideas whats going on or is there a step I am missing when trying to build React-Aria locally?

**Comment by snowystinger** [comment]
Wouldn't you need to link/install every react-aria package if you were doing this?
I would guess you've ended up with duplicates of some packages, check your lock file or run something like https://github.com/adobe/react-spectrum/wiki/Frequently-Asked-Questions-(FAQs)#why-are-there-errors-after-upgrading-a-react-spectrum-package

**Reply by bu3alwa** [reply]
```
    "@react-aria/calendar": "file:../../../react-spectrum/packages/@react-aria/calendar",
    "@react-aria/focus": "file:../../../react-spectrum/packages/@react-aria/focus",
    "@react-aria/i18n": "file:../../../react-spectrum/packages/@react-aria/i18n",
    "@react-aria/utils": "file:../../../react-spectrum/packages/@react-aria/utils",
    "@react-stately/calendar": "file:../../../react-spectrum/packages/@react-stately/calendar",
    "react-aria": "file:../../../react-spectrum/packages/react-aria",
    "react-aria-components": "file:../../../react-spectrum/packages/react-aria-components",
```

So I have these in the package.json. I see that they are linked in the lock file to the correct path. 

So I am guessing I should just add all the packages for it to work?

I was trying linking all of them through workspaces as well but was running into different issues

**Reply by snowystinger** [reply]
right, you might need to add resolutions for all of them as well though, because each one is going to specify versions of everything they depend on, not the local paths

**Reply by bu3alwa** [reply]
hmm is it easier to do workspaces then? or will I run into other issues there?

**Reply by bu3alwa** [reply]
```
  "workspaces": [
    "../../../react-spectrum/packages/@adobe/react-spectrum",
    "../../../react-spectrum/packages/@internationalized",
    "../../../react-spectrum/packages/react-aria",
    "../../../react-spectrum/packages/react-aria/*",
    "../../../react-spectrum/packages/react-aria-components",
    "../../../react-spectrum/packages/react-stately",
    "../../../react-spectrum/packages/react-stately/*",
    "../../../react-spectrum/packages/@react-aria",
    "../../../react-spectrum/packages/@react-aria/*",
    "../../../react-spectrum/packages/@react-types",
    "../../../react-spectrum/packages/@react-icons",
    "../../../react-spectrum/packages/*/*"
  ],
```

So I added these and I am still required to build all the packages. I did remove the dependencies so they don't conflict.

![image](https://github.com/adobe/react-spectrum/assets/3469062/2e424234-5180-49ac-a39e-a50da6cb4ff0)

Side note is it possible to just use the source instead of the dist thats transpiled for debugging?

**Reply by snowystinger** [reply]
When running locally against a different project, I typically debug just in node_modules, though I do have the advantage of knowledge of the code base. Otherwise, I run verdaccio and publish all the packages there, then point my npmrc at it and install from there as well.

I'm not sure about the differences between using workspaces vs verdaccio. However, for some of our examples apps, we do have workspaces set up like so https://github.com/adobe/react-spectrum/blob/main/examples/next-app/package.json

**Reply by bu3alwa** [reply]
Yea I have the workspace setup following the example. And I also have the resolutions for react there too. But it seems to be running into that null error above. I don't understand why it runs into it.

**Reply by snowystinger** [reply]
Have you checked to make sure there's only one copy of each of our packages? The link I shared in my first comment has a script that should make that easy to check.

Otherwise, I'd check if you're loading two different files from the same package, ie cjs and mjs. You can check that by putting console logs at the root of each file in a package you suspect might be doing that.

**Reply by bu3alwa** [reply]
Yea the script yielded no results in the report so I assume thats means its fine there are no errors? One thing I noticed even with the resolution that when the react aria workspace calls react its pointing to `../react-spectrum/node_modules/react/cjs/react.development.js` I am assuming this might be the issue but I don't understand why the resolutions on my package is not overriding react being called in react aria.

**Reply by snowystinger** [reply]
that seems like a solid lead, i'm not sure why that would be either
maybe you can delete the node_modules in `../react-spectrum` after performing the build? what version of yarn are you using? you could try a different version

**Reply by bu3alwa** [reply]
I am giving up on this method. I tried pnpm npm and yarn and they all have different methods and different issues to get them to override. I am going to go with verdaccio
